rclcpp/src/rclcpp/init_options.cpp

```diff
@@ -99,4 +99,36 @@ InitOptions::get_rcl_init_options() const
   return init_options_.get();
 }
 
+void
+InitOptions::use_default_domain_id()
```

Comment:
It sounds like all setters and getters in this class aren't thread safe.
It would be good to address that problem in another PR, maybe before merging this one

Reply:
correct, this needs to be fixed before merging this one.